### ISSUE TITLE
No longer working with Plasma 6 Wayland

### ISSUE THREAD

**Issue by HanPrower** [xremap/xremap#433]
I've had xremap setup and working on Arch Linux before on Plasma Wayland 5.2x so it's configured correctly, but it's just existed for a while without me needing it, so it's possible it stopped working earlier...

So, xremap still seems to work in-so-much as it runs, can find the devices, even remap keys if the only application is set to `not`, but then when it attempts to connect the KDEClient to detect the Active Window it to fails.

`Mar 08 18:55:40 smokey systemd-logind[616]: Watching system buttons on /dev/input/event22 (xremap)`

2 minutes later...

`Mar 08 18:57:54 smokey xremap[28122]: application-client: KDE (supported: false)`

It seems to be failing around [Line 121 of kde_client.rs](https://github.com/k0kubun/xremap/blob/master/src/client/kde_client.rs#L121). I guess it's failing to create `com.k0kubun.Xremap`? I don't see it in QDBusViewer... but I don't know exactly what should be happening.

Happy to debug and try to fix it with guidance :)

**Comment by HanPrower** [comment]
Thanks, rebuilding with those changes does indeed let the client start and report window changes.

Although for some reason it can fail multiple times until the `application-client` starts successfully. Not sure if that's a Plasma 6 issue or a me issue...

**Comment by k0kubun** [comment]
Hmm. Should I push that change then? It seems hard to support multiple KDE versions in that case though :confused: 

@N4tus As the author of https://github.com/k0kubun/xremap/pull/264, WDYT?

**Comment by Ahrotahn** [comment]
If you don't care about adding another release target then an alternative option for supporting both would be to split the `kde` feature into separate `kde5` and `kde6` features.  Then the changes can be gated by the feature used.

**Comment by N4tus** [comment]
First of, thank you @Ahrotahn for figuring out what api changed. This would have taken me countless hours to discover myself, and I truly appreciate it.

My original idea was to have another back-end specifically for plasma 6. So that a kde and kde6 back-end could live side by side and the user would have to choose the one they need. But if we only need to change the object-path and load a different script/patch the existing script depending on the version, my idea might be not as good.

Is the commit, that changes the script-path, part of plasma 6 or earlier?  Because I am wondering if #393 was caused by this change, which would mean that the version of plasma would not be enough to determine what object path to use. In this case we would also need to know who packaged plasma, which would complicate it. If this is the case we also may need a separate check, whether to use `clientActivated` or `windowActivated`. I'm asking these questions because this issue suggests that is is already broken on plasma 5.2x. But I'm running xremap on fedora 39 with plasma 5.27.10 on wayland without issues. This makes me think that there is definitely something fishy going on here.

In all cases I think that the old version of this back-end should be preserved somehow, since not everyone is using the latest version of kde and they would then be stuck on an old version of xremap.

**Comment by k0kubun** [comment]
Maybe we could try one path first and then move on to the other path when it returns an error?

**Comment by N4tus** [comment]
#437 should have brought support for kde 6, so I am closing this issue. In case stuff does not work, we can reopen or create a new issue.
